jetstream_pt/engine.py

```diff
@@ -280,19 +307,119 @@ def _call_model_prefill(self, weights, tokens, input_indexes):
     caches_res = [c.state() for c in caches]
     return torchjax.from_torch((res, caches_res))
 
-  def _sampling(self, logits: Any, batch_size: int) -> jnp.ndarray:
+  def _greedy_sampling(self, logits):
+    return jnp.argmax(logits, axis=-1)
+
+  def _weighted_sampling(self, logits, rng, temperature):
+    return jax.random.categorical(rng, logits / temperature)
+
+  def _nucleus_sampling(self, logits, rng, temperature, nucleus_topp):
```

Comment:
Should we replace 'nucleus sampling' with 'top-p sampling'? The latter is easier to understand because the name clearly refers the top p.